solr/core/src/test/org/apache/solr/TestDistributedGrouping.java

```diff
@@ -425,6 +426,65 @@ public void test() throws Exception {
     
     //Debug
     simpleQuery("q", "*:*", "rows", 10, "fl", "id," + i1, "group", "true", "group.field", i1, "debug", "true");
+    doTestGroupSkipSecondStep();
+  }
+
+  /*
+    SOLR-11831, test skipping the second grouping step if the query only retrieves on document per group
+   */
+  private void doTestGroupSkipSecondStep() throws Exception {
+    ignoreException(GroupParams.GROUP_SKIP_DISTRIBUTED_SECOND); // don't print stack trace for exception raised by group.skip.second.step
+    // Ignore numFound if group.skip.second.step is enabled because the number of documents per group will not be computed (will default to 1)
```

Comment:
> ... // Ignore numFound if group.skip.second.step is enabled because the number of documents per group will not be computed (will default to 1) ...

Curiosity only at this point, might it be possible to (somehow) test for the `numFound` being returned to be `1`

And as you already mentioned elsewhere in this pull request, the `numFound` always being `1` for `group.skip.second.step=true` needs to be clearly documented.

Reply:
added a check abf38b0ef4a3e08eae0833d3bd6469a9a1187b1c